kpops/utils/cli_commands.py

```diff
@@ -0,0 +1,77 @@
+import logging
+from pathlib import Path
+from typing import Any
+
+import yaml
+from pydantic.fields import FieldInfo
+from pydantic_core import PydanticUndefined
+
+from hooks.gen_docs.gen_docs_env_vars import collect_fields
+from kpops.config import KpopsConfig
+from kpops.utils.docstring import describe_object
+from kpops.utils.json import is_jsonable
+from kpops.utils.yaml import touch_yaml_file
+
+log = logging.getLogger("cli_commands_utils")
+
+
+def extract_config_fields_for_yaml(
+    fields: dict[str, Any], required: bool
+) -> dict[str, Any]:
+    """Return only (non-)required fields and their respective default values.
+
+    :param fields: Dict containing the fields to be categorized. The key of a
+        record is the name of the field, the value is the field's type.
+    :param required: Whether to extract only the required fields or only the
+        non-required ones.
+    """
+    extracted_fields = {}
+    for key, value in fields.items():
+        if issubclass(type(value), FieldInfo):
+            if required and value.default in [PydanticUndefined, Ellipsis]:
+                extracted_fields[key] = None
+            elif not (required or value.default in [PydanticUndefined, Ellipsis]):
+                if is_jsonable(value.default):
```

Comment:
does that work if the default value is an instance of `pydantic.BaseModel`?

Reply:
Good point, it appears that it doesnt

```python
>>> from pydantic import BaseModel, Field
>>> from kpops.utils.json import is_jsonable
>>> class A(BaseModel):
...     foo: str
...
>>> class B(BaseModel):
...     bar: A = Field(default=A(foo="foo"))
...
>>> is_jsonable(B())
False
>>> str(B())
"bar=A(foo='foo')"
```

Reply:
Added a case that covers it